components/forms/billing-form-new-button.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import { useTransition } from "react";
+import { generateUserStripe } from "@/actions/generate-user-stripe";
+import { SubscriptionPlan, UserSubscriptionPlan } from "@/types";
+
+import { Button } from "@/components/ui/button";
+import { Icons } from "@/components/shared/icons";
+
+interface BillingFormButtonProps {
+  offer: SubscriptionPlan;
+  subscriptionPlan: UserSubscriptionPlan;
+  year: boolean;
+}
+
+export function BillingFormNewButton({
+  year,
+  offer,
+  subscriptionPlan,
+}: BillingFormButtonProps) {
+  let [isPending, startTransition] = useTransition();
+  const generateUserStripeSession = generateUserStripe.bind(
+    null,
+    offer.stripeIds[year ? "yearly" : "monthly"],
+  );
+
+  const stripeSessionAction = () =>
+    startTransition(async () => await generateUserStripeSession());
+
+  const userOffer =
+    subscriptionPlan.stripePriceId ===
+    offer.stripeIds[year ? "yearly" : "monthly"];
+
+  return (
+    <Button
+      variant={userOffer ? "default" : "outline"}
+      rounded="full"
+      className="w-full"
+      disabled={isPending}
+      onClick={stripeSessionAction}
+    >
+      {isPending ? (
+        <>
+          <Icons.spinner className="mr-2 size-4 animate-spin" /> Loading...
+        </>
+      ) : (
+        <>{userOffer ? "Manage Subscription" : "Upgrade"}</>
+      )}
```

Comment:
@echolayer can you remove the isPending conditional and just show the result without the Loading state